packages/common/src/gckms/KeyInjectorPlugin.js

```diff
@@ -0,0 +1,41 @@
+const kms = require("@google-cloud/kms");
+const { Storage } = require("@google-cloud/storage");
+const { extendConfig } = require("hardhat/config");
+const { HardhatPluginError } = require("hardhat/plugins");
+const { getGckmsConfig } = require("./GckmsConfig");
+
+const { GCKMS_KEYS, MNEMONIC } = process.env;
+
+async function retrieveGckmsKeys(gckmsConfigs) {
+  return await Promise.all(
+    gckmsConfigs.map(async (config) => {
+      const storage = new Storage();
+      const keyMaterialBucket = storage.bucket(config.ciphertextBucket);
+      const ciphertextFile = keyMaterialBucket.file(config.ciphertextFilename);
+
+      const contentsBuffer = (await ciphertextFile.download())[0];
+      const ciphertext = contentsBuffer.toString("base64");
+
+      // Send the request to decrypt the downloaded file.
+      const client = new kms.KeyManagementServiceClient();
+      const name = client.cryptoKeyPath(config.projectId, config.locationId, config.keyRingId, config.cryptoKeyId);
+      const [result] = await client.decrypt({ name, ciphertext });
+      return "0x" + Buffer.from(result.plaintext, "base64").toString().trim();
+    })
+  );
+}
+
+extendConfig((config) => {
```

Comment:
to double-check: this is the synchronous call that you were talking about offline, where and the part that cant be async?

Reply:
Yep, exactly. As far as I can tell, there's no way to make this async. Another setup that would work just as well, but may be a bit less explicit would be to make the entire method async and use `await` syntax throughout. This would work, but not because hardhat supports it, just because it happens to finish before hardhat needs the result.